helix-term/src/args.rs

```diff
@@ -41,15 +42,51 @@ impl Args {
                         }
                     }
                 }
-                arg => args.files.push(PathBuf::from(arg)),
+                arg => args.files.push(parse_file(arg)),
             }
         }
 
         // push the remaining args, if any to the files
-        for filename in iter {
-            args.files.push(PathBuf::from(filename));
+        for arg in iter {
+            args.files.push(parse_file(arg));
         }
 
         Ok(args)
     }
 }
+
+/// Parse arg into [`PathBuf`] and position.
+pub(crate) fn parse_file(s: &str) -> (PathBuf, Position) {
+    split_path_row_col(s)
+        .or_else(|| split_path_row(s))
+        .unwrap_or_else(|| (PathBuf::from(s), Position::default()))
+}
+
+/// Split file.rs:10:2 into [`PathBuf`], row and col.
+///
+/// Does not validate if file.rs is a file or directory.
+fn split_path_row_col(s: &str) -> Option<(PathBuf, Position)> {
+    if Path::new(s).exists() {
+        return None;
+    }
```

Comment:
Seems like this should go in `parse_file` instead to limit duplication of code.

Reply:
Thanks, I missed this.